### ISSUE TITLE
Sequence should be per Entity

### ISSUE THREAD

**Issue by simasch** [vaadin/starters#102]
I like the move from UUID to Long, but now there is only one SequenceGenerator because the ID is defined in the AbstractEntity.

That's not good practice, and I've never seen this in real-world-projects. Usually, there is one sequence per table.

This code should be moved to the Entities:
```java
@Id
@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "idgenerator")
// The initial value is to account for data.sql demo data ids
@SequenceGenerator(name = "idgenerator", initialValue = 1000)
private Long id;
```

**Comment by Artur-** [comment]
Hmm, https://hibernate.atlassian.net/browse/HHH-14492 claims that Hibernate 6 should use per-entity sequences by default

**Comment by simasch** [comment]
Yes, but this is also very limited as you cannot specify the name or the allocationSize.

The default will kick in if you don't specify a generator:

    @GeneratedValue(strategy = GenerationType.SEQUENCE)
